test/distributed/org/apache/cassandra/distributed/test/streaming/BoundExceptionTest.java

```diff
@@ -98,46 +91,26 @@ public void testEmptyStackTrace()
         Throwable exceptionToTest = new NullPointerException("there are words here");
         exceptionToTest.setStackTrace(new StackTraceElement[0]);
 
-        Throwable boundedStackTrace = StreamSession.boundStackTrace(exceptionToTest, LIMIT);
+        StringBuilder boundedStackTrace = StreamSession.boundStackTrace(exceptionToTest, LIMIT);
         String expectedStackTrace = "java.lang.NullPointerException: there are words here\n";
 
-        assertEquals(expectedStackTrace,Throwables.getStackTraceAsString(boundedStackTrace));
-        assertEquals(boundedStackTrace.getStackTrace().length, 0);
-    }
-
-    @Test
-    public void testLimitLargerThanStackTrace()
-    {
-        Throwable exceptionToTest = new NullPointerException("there are words here");
-        Throwable boundedStackTrace = StreamSession.boundStackTrace(exceptionToTest, LIMIT);
-        Throwable reboundedStackTrace = StreamSession.boundStackTrace(boundedStackTrace, LIMIT + 2);
-
-        assertEquals(reboundedStackTrace, boundedStackTrace);
-        assertEquals(reboundedStackTrace.getStackTrace().length, LIMIT);
+        assertEquals(expectedStackTrace,boundedStackTrace.toString());
     }
 
     @Test
     public void testEmptyNestedStackTrace()
     {
         Throwable exceptionToTest = new RuntimeException(new IllegalArgumentException("the disk /foo/var is bad", new IOException("Bad disk somewhere")));
-
         exceptionToTest.setStackTrace(new StackTraceElement[0]);
-        exceptionToTest.getCause().setStackTrace(new StackTraceElement[0]);
         exceptionToTest.getCause().getCause().setStackTrace(new StackTraceElement[0]);
 
-        Throwable boundedStackTrace = StreamSession.boundStackTrace(exceptionToTest, LIMIT);
+        StringBuilder boundedStackTrace = StreamSession.boundStackTrace(exceptionToTest, LIMIT);
         String expectedStackTrace = "java.lang.RuntimeException: java.lang.IllegalArgumentException: the disk /foo/var is bad\n" +
-                                     "Caused by: java.lang.IllegalArgumentException: the disk /foo/var is bad\n" +
-                                     "Caused by: java.io.IOException: Bad disk somewhere\n";
-
-        String boundedStackTraceAsString = Throwables.getStackTraceAsString(boundedStackTrace);
-
-        assertEquals(boundedStackTraceAsString, expectedStackTrace);
+                                    "java.lang.IllegalArgumentException: the disk /foo/var is bad\n" +
+                                    "\torg.apache.cassandra.distributed.test.streaming.BoundExceptionTest.testEmptyNestedStackTrace(BoundExceptionTest.java:103)\n" +
```

Comment:
I am ok with this semantic if that is what you feel is best, as we only print 2 elements... left a comment in another test that you are printing more than 2 frames and that middle causes are less important.